File: DotNetflixServer/Domain/Entities/Subscription.cs
namespace Domain.Entities;

public class Subscription
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public int Cost { get; set; }
    public int? PeriodInDays { get; set; }
    public bool IsAvailable { get; set; }
    public List<MovieInfo> Movies { get; set; } = null!;
    public List<User> Users { get; set; } = null!;
    public List<UserSubscription> UserSubscriptions { get; set; } = null!;
    public List<SubscriptionMovieInfo> SubscriptionMovies { get; set; } = null!;
}

File: mobile/lib/models/subscription.dart
class Subscription {
  final int id;
  final String name;
  final int cost;
  final int? periodInDays;
  final List<String> filmNames;
  bool belongsToUser;

  Subscription.fromDynamic(dynamic object)
    : id = object['id'],
      name = object['name'],
      cost = object['cost'],
      periodInDays = object['periodInDays'],
      filmNames = (object['filmNames'] as List).map((x) => x.toString()).toList(),
      belongsToUser = object['belongsToUser'];
}
